### PR TITLE
feat(EMS-3977): broker details - form validation updates

### PR DESCRIPTION
**Pull request by ttbarnes**
## Introduction :pencil2:
This PR updates the form validation in the "Broker details" form to include validation rules for the following fields:
- Based in the UK
- Postcode
- Building number/name

Note: E2E test coverage for postcode field will be improved in a separate ticket PR.

## Resolution :heavy_check_mark:
- Update `completeBrokerDetailsForm` cypress command.
- Update `MAXIMUM_CHARACTERS` constants.
- Update error messages content strings.
- Update E2E application fixtures.
- Update `BROKER_DETAILS` validation E2E test.
- Update API schema, SQL dump.
- Create new UI `BROKER_DETAILS` validation rules.
- Update nunjucks template to consume submitted values.

## Miscellaneous :heavy_plus_sign:
- Rename instances of `postCode` to `postcode`.

### CODE REVIEW

File: e2e-tests/content-strings/error-messages.js [hunk 1]
```diff
@@ -375,6 +375,17 @@ export const ERROR_MESSAGES = {
           ABOVE_MAXIMUM: `The broker or company email cannot be more than ${MAXIMUM_CHARACTERS.EMAIL} characters`,
           INCORRECT_FORMAT: 'Enter the broker or company email address in the correct format, like name@example.com',
         },
+        [FIELD_IDS.INSURANCE.POLICY.BROKER_DETAILS.IS_BASED_IN_UK]: {
+          IS_EMPTY: 'Select if the broker you are using in the UK',
+        },
+        [FIELD_IDS.INSURANCE.POLICY.BROKER_DETAILS.POSTCODE]: {
+          IS_EMPTY: "Enter the broker's post code",
```
Comment (Zainzzkk) [inline e2e-tests/content-strings/error-messages.js:382]: this line says post code, line belows says postcode.  should it be standardised?

File: src/ui/server/content-strings/error-messages.ts [hunk 2]
```diff
@@ -379,6 +379,17 @@ export const ERROR_MESSAGES = {
           ABOVE_MAXIMUM: `The broker or company email cannot be more than ${MAXIMUM_CHARACTERS.EMAIL} characters`,
           INCORRECT_FORMAT: 'Enter the broker or company email address in the correct format, like name@example.com',
         },
+        [FIELD_IDS.INSURANCE.POLICY.BROKER_DETAILS.IS_BASED_IN_UK]: {
+          IS_EMPTY: 'Select if the broker you are using in the UK',
+        },
+        [FIELD_IDS.INSURANCE.POLICY.BROKER_DETAILS.POSTCODE]: {
+          IS_EMPTY: "Enter the broker's post code",
```
Comment (Zainzzkk) [inline src/ui/server/content-strings/error-messages.ts:386]: same as above

File: src/ui/server/controllers/insurance/policy/broker-details/index.test.ts [hunk 3]
```diff
@@ -177,7 +179,7 @@ describe('controllers/insurance/policy/broker-details', () => {
         await post(req, res);
       });
 
-      it(`should redirect to ${BROKER_CONFIRM_ADDRESS_ROOT}`, () => {
+      it.only(`should redirect to ${BROKER_CONFIRM_ADDRESS_ROOT}`, () => {
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/broker-details/index.test.ts:182]: should this only be here?

File: src/ui/server/controllers/insurance/policy/broker-details/validation/rules/postcode/index.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const {
+  BROKER_DETAILS: { [FIELD_ID]: ERROR_MESSAGE },
+} = ERROR_MESSAGES.INSURANCE.POLICY;
+
+/**
+ * validate the "broker postcode" field
+ * @param {RequestBody} formBody: Form body
+ * @param {Object} errors: Errors from previous validation errors
+ * @returns {ValidationErrors} postcodeValidation
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/broker-details/validation/rules/postcode/index.ts:18]: typo needs space and no capital v?

Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/policy/broker-details/validation/rules/postcode/index.ts:18]: should this say result of postcodeValidation?

File: e2e-tests/content-strings/error-messages.js (hunk 1 above)
Comment (Zainzzkk) [reply e2e-tests/content-strings/error-messages.js:382]: sorry I mean the text for the error.  IS_EMPTY says post code and INCORRECT_FORMAT says postcode

**Review by Zainzzkk — APPROVED** [review]

**Comment by ttbarnes** [comment]
Merging this with some failing tests (unrelated to this PR), to help clear the PR backlog. Other PRs have fixes.

**Pull request merged**